src/network/message_blockdata.rs

```diff
@@ -43,6 +43,20 @@ pub enum Inventory {
     WitnessBlock(BlockHash),
 }
 
+impl Inventory {
+    /// Return the item value represented as a SHA256-d hash.
+    pub fn hash(&self) -> sha256d::Hash {
+        match *self {
+            Inventory::Error => sha256d::Hash::default(),
```

Comment:
I feel a bit uneasy about returning an all-zero "hash". I don't know how exactly you'd misuse this in a dangerous manner that isn't already misguided without that fact but there might exist one. Would it be too bad to return a `Result<Hash, Error>`? Or is this just stupid in this context (I'm not entirely sure what you want to use the fn for)?

Reply:
I think it can be done through `Option<sha256d::Hash>` return type

Reply:
Yeah, right, the error type would only have a single instance anyway.

Reply:
Hmm, fair enough, though in the protocol, the Error variant is represented by 32 0-bytes as well. So maybe "hash" isn't the right name in that case, maybe we'd do an `as_bytes() -> [u8; 32]` even though using `sha256d::Hash` is just so much more convenient. Not to speak about hex representation and it's reversal problems etc.